sdr_repository.go

```diff
@@ -59,12 +60,24 @@ func RetrieveSDRRepository(ctx context.Context, s Session) (SDRRepository, error
 
 // walkSDRs iterates over the SDR Repository. It is not concerned with the repo
 // changing behind its back.
```

Comment:
> It is not concerned with the repo changing behind its back.

I think this is still true (i.e. we still need the two `GetSDRRepositoryInfo` requests) because section 33.11.2 of IPMI v2.0 implies that adding or deleting an SDR does not necessarily change other records IDs.

Additionally,  section 33.11.1 states that:
> since Record IDs could change between offset 0 ‘Gets’ of a given record, it is the responsibility of the
device accessing the repository to verify that the retrieved record information matches up with the ID
information